examples/src/components/DateFilterComponentExample.jsx

```diff
@@ -0,0 +1,215 @@
+        ],
+        "GDC.time.quarter": [
+            {
+                from: 0,
+                to: 0,
+                granularity: "GDC.time.quarter",
+                localIdentifier: "THIS_QUARTER",
+                type: "relativePreset",
+                visible: true,
+                name: "",
+            },
+            {
+                from: -1,
+                to: -1,
+                granularity: "GDC.time.quarter",
+                localIdentifier: "LAST_QUARTER",
+                type: "relativePreset",
+                visible: true,
+                name: "",
+            },
+            {
+                from: -3,
+                to: 0,
+                granularity: "GDC.time.quarter",
+                localIdentifier: "LAST_4_QUARTERS",
+                type: "relativePreset",
+                visible: true,
+                name: "",
+            },
+        ],
+        "GDC.time.year": [
+            {
+                from: 0,
+                to: 0,
+                granularity: "GDC.time.year",
+                localIdentifier: "THIS_YEAR",
+                type: "relativePreset",
+                visible: true,
+                name: "",
+            },
+            {
+                from: -1,
+                to: -1,
+                granularity: "GDC.time.year",
+                localIdentifier: "LAST_YEAR",
+                type: "relativePreset",
+                visible: true,
+                name: "",
+            },
+        ],
+    },
+};
+
+export class DateFilterComponentExample extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedFilterOption: defaultDateFilterOptions.allTime,
+            excludeCurrentPeriod: false,
+        };
+    }
+
+    onApply = (dateFilterOption, excludeCurrentPeriod) => {
+        this.setState({
+            selectedFilterOption: dateFilterOption,
+            excludeCurrentPeriod,
+        });
+        // eslint-disable-next-line no-console
+        console.log(
+            "DateFilterExample onApply",
+            "selectedFilterOption:",
+            dateFilterOption,
+            "excludeCurrentPeriod:",
+            excludeCurrentPeriod,
+        );
+    };
+
+    onCancel = () => {
+        // eslint-disable-next-line no-console
+        console.log("DateFilterExample onCancel");
+    };
+
+    render() {
+        return (
+            <div style={{ width: 300 }}>
+                <DateFilter
+                    excludeCurrentPeriod={this.state.excludeCurrentPeriod}
+                    selectedFilterOption={this.state.selectedFilterOption}
+                    filterOptions={defaultDateFilterOptions}
+                    availableGranularities={availableGranularities}
+                    customFilterName="Selected date"
+                    dateFilterMode="active"
+                    onApply={this.onApply}
+                    onCancel={this.onCancel}
+                    onOpen={this.onOpen}
```

Comment:
Seems to me like `this.onOpen` and `this.onClose` are undefined here.

Reply:
I created callbacks with console logs. Closing